flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/StatusHelper.java

```diff
@@ -64,6 +67,14 @@ public <T extends CustomResource<?, STATUS>> void patchAndCacheStatus(T resource
         // in the meantime
         resource.getMetadata().setResourceVersion(null);
 
+        ObjectNode newStatus = objectMapper.convertValue(resource.getStatus(), ObjectNode.class);
+        ObjectNode previousStatus = statusCache.put(getKey(resource), newStatus);
+
+        if (previousStatus == null || newStatus.equals(previousStatus)) {
```

Comment:
why here do not need to patch status when the previousStatus is null ?

Reply:
Very good catch you are right @Aitozi